protos/flyteidl/core/security.proto

```diff
@@ -43,6 +43,11 @@ message Secret {
     // will depend on the key management system.
     // +optional
     MountType mount_requirement = 4;
+
+    // The name of the environment variable, if the Secret is injected as environment variable. If ommitted, the default
+    // FLYTE_SECRETS_ENV_PREFIX prefix will be used.
+    // +optional
+    string env_name = 5;
```

Comment:
This only applies to `ENV_VAR` mount type. I would make the change as following:

1. Mark `MountType mount_requirement = 4 [deprecated=true];` as deprecated.
2. Add:
    ```protobuf=
      oneof MountTarget mount_target {
        MountEnvVar env_var = 5;
        MountFile file = 6;
      }

      message MountEnvVar {
        string name = 1;
      }

      message MountFile {
        string path = 1;
      }
    ```
3. Update secrets webhook to look at the new field first, if it's nil, it should fallback to the old field.
4. Update flytekit to always set both fields; `mount_requirement` and `mount_target`

Reply:
Thanks for the feedback! I'll try to add it. Could you explain what you mean with 4? Not sure I get what you mean here.

Reply:
Added the changes with a working initial version, I'll get to adding a number of tests as well.

Reply:
Tests added, let me know what you think :)

Reply:
I've opened PRs here for the change, I'll close this one.

- https://github.com/flyteorg/flytekit/pull/1726
- https://github.com/flyteorg/flytepropeller/pull/589
- https://github.com/flyteorg/flyteidl/pull/423
- https://github.com/flyteorg/flytesnacks/pull/1017